lib/Middleware/SessionMiddleware.php

```diff
@@ -116,7 +119,7 @@ private function assertUserOrShareToken(ISessionAwareController $controller): vo
 		$documentId = (int)$this->request->getParam('documentId');
 		if (null !== $userId = $this->userSession->getUser()?->getUID()) {
 			// Check if user has access to document
-			if (count($this->rootFolder->getUserFolder($userId)->getById($documentId)) === 0) {
+			if ($this->rootFolder->getUserFolder($userId)->getFirstNodeById($documentId) === null) {
```

Comment:
Originally from https://github.com/nextcloud/text/pull/5933 - included here for consistency (we also use `getFirstNodeById` below).

Reply:
Reverted as Nc 28 does not have `getFirstNodeById`